### ISSUE TITLE
Not compatible with Thunderbird 68.2.1

### ISSUE THREAD

**Issue by paddatrapper** [snakelizzard/flexible_identity#1]
After upgrading to Thunderbird 68.2.1, this extension was disabled with the message "Flexible Identity is incompatible with Thunderbird 68.2.1"

See https://support.mozilla.org/en-US/kb/unable-install-add-on-extension-theme-thunderbird

**Comment by enrique-henri** [comment]
I noticed this too.
How great it would be if it was comaptible as I have been using it for several years.
I'll keep my fingers crossed but won't hold my breath.

**Comment by arrfab** [comment]
Yep, was using that extension and sad that it's not compatible ..

**Comment by nneul** [comment]
Anyone checked to see if this is "actually incompatible due to API changes" or just "not marked as compatible in version range"?

**Comment by nneul** [comment]
Definitely has to have changes - to new json manifest setup. Unclear if there are secondary changes as I'm not a TB developer.

**Comment by snakelizzard** [comment]
Just to state what I have tried. I migrated to json config, but it does not work for me. Error console in Thunderbird does not show any hint on what I doing wrong. Next thing to do is to create simple empty extension and migrate files there one by one starting with overlays. Sorry guys, it’s been a lot of trouble lately to concentrate on this

**Comment by nneul** [comment]
Thank you for your efforts! This is definitely one of those 'can't live without' extensions for TB. Do you have your WIP on a branch to look at? (or can you attach the updated config)?

**Comment by snakelizzard** [comment]
Here is sample in 25ae2f4
Most likely reason why is not working descried in section 'Notes about overlaying' [here](https://developer.thunderbird.net/add-ons/tb68/overlays#notes-about-overlaying-in-general)
Mozilla issue to track [is 1476259](https://bugzilla.mozilla.org/show_bug.cgi?id=1476259)

**Comment by FredThompsonII** [comment]
TBird 68.12.0 reports the same error with Flexible Identity 1.0.6...sigh...any progress on squashing this bug?

**Comment by snakelizzard** [comment]
@FredThompsonII Hello. Nope, I give up supporting this addon.

**Comment by Yonn-Trimoreau** [comment]
Have you heard of any similar extension that does work on Thunderbird 68?
That's also a must have for me, I can't imagine working without it... I have almost 10 different identities...

**Comment by snakelizzard** [comment]
@Yonn-Trimoreau Yeah, I honestly don't know what to recommend. Maybe keep the old version of Thunderbird. For me, the proportion of correspondence via email decreased noticeably years ago, same for number of identities. I settled on a separate virtual machine for each working context with specific software configured in each. So I can't even tell you what addons are out there now.

**Comment by Yonn-Trimoreau** [comment]
It's a shame that you had to virtualize your environment to be able to work on multiple contexts... Maybe it's more practical to you, but for my part, in terms of performance, that's simply not feasible.

Any way, I finally installed two extensions:
- https://github.com/speedball2001/identitychooser-mx
- https://github.com/dreadnaut/bordercolors-d

The functionalities are not exactly the same, but it greatly helps me working with my multiple identities.

I had to keep TB from upgrading to 68 for 2 years, and the only reason was your extension. Your work was really great. Sad that you stopped maintaining it and nobody took over.

**Comment by schoettl** [comment]
In case someone ever forks und upgrades this plugin to the new WebExtension Plugin API (for Thunderbird > 60):

There is a before-send hook! https://webextension-api.thunderbird.net/en/91/compose.html#onbeforesend

I guess that is the key function; the rest is "just" porting the nice user interface of Flexible Identity to the WebExtension API.

PS: This [fork of Correct Identity](https://github.com/ldreier/tb-correctidentity) works with newer Thunderbird and is a good alternative!
